components/reusable-components/navigation/Navigation.tsx

```diff
@@ -3,53 +3,139 @@
 
 'use client';
 
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import Link from 'next/link';
 import Image from 'next/image';
-import 'bootstrap-icons/font/bootstrap-icons.css';
-
-import styles from './navigation.module.scss';
-import { useTheme } from '../../../app/context/themeContext';
+import { usePathname } from 'next/navigation';
+import { FaUser, FaSignOutAlt, FaChevronDown, FaChevronUp, FaEnvelope } from 'react-icons/fa';
 import sunImage from '../../../public/theme/sun.png';
 import moonImage from '../../../public/theme/moon.png';
 import LogoNavigation from '../../../public/logo/logo-black.svg';
+import styles from './navigation.module.scss';
+import { useTheme } from '../../../app/context/themeContext';
+import { User } from '../_Types';
+
+const user: User = {
+  name: 'John Doe',
+  email: 'john@example.com',
+  avatar: '/user.png',
+};
 
-const Navigation = () => {
+const Navigation: React.FC = () => {
   const { theme, toggleTheme } = useTheme();
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
+  const menuRef = useRef<HTMLDivElement | null>(null);
+  const pathname = usePathname();
+
+  const isContentPanel = pathname.startsWith('/content-panel');
   const isSun = theme === 'light';
 
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (menuRef.current && !menuRef.current.contains(event.target as Node)) {
+        setIsMenuOpen(false);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => document.removeEventListener('mousedown', handleClickOutside);
+  }, []);
```

Comment:
Can you create a custom hook for this and place it in the utils folder?

Reply:
utils/HOC/useClickOutside created